libvmi/driver/xen/xen_events.c

```diff
@@ -862,7 +862,10 @@ status_t xen_set_mem_access(vmi_instance_t vmi, mem_access_event_t *event,
     dbprint(VMI_DEBUG_XEN, "--Setting memaccess for domain %"PRIu64" on physical address: %"PRIu64" npages: %"PRIu64"\n",
         dom, event->physical_address, npages);
 
-    rc = xc_set_mem_access(xch, dom, access, page_key, npages);
+    if ( !altp2m_idx )
```

Comment:
0 is treated specially by Xen as the base pagetable. All other altp2m views take 0 as the base and get populated lazily in a copy-on-write fashion when using xc_altp2m_set_mem_access. So for 0 you should keep using the same xc_set_mem_access routines as there is no difference for that view from a mem_access perspective.

Some documentation is available at http://lists.xenproject.org/archives/html/xen-devel/2015-06/msg01319.html